### DISCUSSION TITLE
Where condition on column not working for 'equal' cases

### DISCUSSION THREAD

**Discussion by PierreLucB** [ujjwalguptaofficial/JsStore#270] [General]
Hello,
I can't succeed to retrieve a value in a table while directly filtering the column by his value.

Here is my code : 
```
declare var require: any;

const getWorkerPath = () => {
    if (environment.production == false) {
         return require("file-loader?name=scripts/[name].[hash].js!jsstore/dist/jsstore.worker.js");
    }
    else {
       return require("file-loader?name=scripts/[name].[hash].js!jsstore/dist/jsstore.worker.min.js");
    }
};
const workerPath = getWorkerPath().default;

export const conForTests = new Connection(new Worker(workerPath));

  // Create db with initial table
      conForTests.initDb({
         name: 'DbTests', tables: [{
            name: 'Table1',
            columns: {
               col1: { primaryKey: true, keyPath: ['col1'] }
            }
         }]
      }).then(() => {

         // value for tests
         const col1valeur = '2';

         // Insert the value
         conForTests.insert({ into: 'Table1', values: [{ col1: col1valeur }] })
            .then(res => {
               console.log('Should be 1 : ' + res);

               // Retrieve the value => doesn't works
               conForTests.select({ from: 'Table1', where: { col1: col1valeur } }).then(res => {
                  console.log('value found', res);
               });

               // Retrieve the value with a 'like' operator => works
               conForTests.select({ from: 'Table1', where: { col1: { like: col1valeur } } }).then(res => {
                  console.log('value found 2 ', res);
               });
            });
      })

```

The where condition isn't working in the first case, I have to use the like operator to filter the values.

Please tell me what am I missing.

Thx

**Comment by ujjwalguptaofficial** [comment]
is your value stored as number or string ?

**Comment by PierreLucB** [comment]
It's in string

**Comment by ujjwalguptaofficial** [comment]
Can't say what's happening, it should work but my guess is there is something wrong with data - may be some space, or other chracters.

Can you provide me some reproducable please ?

**Comment by PierreLucB** [comment]
the code I provided can works as standalone. The value "2" of the variable col1valeur cannot be found

**Comment by ujjwalguptaofficial** [comment]
Hey @PierreLucB 

I tried in idbstudio for numbers and it is working for me.
![Screenshot from 2022-06-07 09-03-55](https://user-images.githubusercontent.com/10938615/172290702-0ac27df1-3db8-4821-88c5-fbfbf2cc3602.png)
![Screenshot from 2022-06-07 09-08-50](https://user-images.githubusercontent.com/10938615/172290763-3385572f-4e4e-4be8-9878-378874fa3172.png)

I will try your code later.

**Comment by ujjwalguptaofficial** [comment]
@PierreLucB try removing the keypath and it should work. Let me know if it does not.

**Comment by PierreLucB** [comment]
Honestly after few days working with jsstore, I just abandoned and made my own solution.

**Comment by ujjwalguptaofficial** [comment]
yeah that's cool :).
